apps/theming-designer/src/components/AccessibilityDetailsList.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import {
+  DetailsList,
+  DetailsRow,
+  IDetailsRowStyles,
+  IDetailsRowProps,
+  IColumn,
+  IGroup,
+  SelectionMode
+} from 'office-ui-fabric-react/lib/DetailsList';
+import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
+import { ITheme } from 'office-ui-fabric-react/lib/Styling';
+import { IContrastRatioPair } from './AccessibilityChecker';
+
+export interface IAccessibilityDetailsListProps {
+  accessiblePairs: IContrastRatioPair[];
+  nonAccessiblePairs: IContrastRatioPair[];
+  theme: ITheme | undefined;
+}
+
+interface IAccessibilityDetailsList {
+  key: string;
+  contrastRatio: String;
+  slotPair: String;
+}
+
+export const AccessibilityDetailsList: React.StatelessComponent<IAccessibilityDetailsListProps> = (
+  props: IAccessibilityDetailsListProps
+) => {
+  let allContrastRatioPairs = props.nonAccessiblePairs.concat(props.accessiblePairs);
+  let nonAccessibleStartIndex = props.nonAccessiblePairs.length;
+
+  let items: IAccessibilityDetailsList[] = [];
+  let groups: IGroup[] = [];
+  let columns: IColumn[] = [];
+  const newTheme = props.theme;
+
+  const onRenderRow = (detailsRowProps: IDetailsRowProps | undefined): JSX.Element => {
+    // Set each row's background and text color to what's specified by its respective slot rule
+    if (detailsRowProps && newTheme) {
+      const currentSlotPair = detailsRowProps!.item.slotPair;
+      const pairSplit = currentSlotPair.split(' on ');
+      const currForegroundColor = pairSplit[0];
+      const currBackgroundColor = pairSplit[1];
+
+      const rowStyles: Partial<IDetailsRowStyles> = {
+        root: {
+          backgroundColor: (newTheme!.palette as any)[currBackgroundColor],
+          color: (newTheme!.palette as any)[currForegroundColor],
+          selectors: {
+            ':hover': {
+              background: 'transparent'
+            }
+          }
+        }
+      };
+      return <DetailsRow {...detailsRowProps!} styles={rowStyles} />;
+    } else {
+      return <div />;
```

Comment:
You could just say `return <div />`, removing the `else` since there is no other action that can be taken.